website/source/docs/providers/aws/r/codebuild_project.html.markdown

```diff
@@ -0,0 +1,158 @@
+---
+layout: "aws"
+page_title: "AWS: aws_codebuild_project"
+sidebar_current: "docs-aws-resource-codebuild-project"
+description: |-
+  Provides n CodeBuild Project resource.
+---
+
+# aws\_codebuild\_project
+
+Provides a CodeBuild Project resource.
+
+## Example Usage
+
+```
+resource "aws_iam_role" "codebuild_role" {
+  name = "codebuild-role-"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "codebuild.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "codebuild_policy" {
+    name        = "codebuild-policy"
+    path        = "/service-role/"
+    description = "Policy used in trust relationship with CodeBuild"
+    policy      = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Resource": [
+        "*"
+      ],
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_iam_policy_attachment" "codebuild_policy_attachment" {
+  name       = "codebuild-policy-attachment"
+  policy_arn = "${aws_iam_policy.codebuild_policy.arn}"
+  roles      = ["${aws_iam_role.codebuild_role.id}"]
+}
+
+resource "aws_codebuild_project" "foo" {
+  name         = "test-project"
+  description  = "test_codebuild_project"
+  timeout      = "5"
+	service_role = "${aws_iam_role.codebuild_role.arn}"
+
+	artifacts {
+		type = "NO_ARTIFACTS"
+	}
+
+  environment {
+    compute_type = "BUILD_GENERAL1_SMALL"
+    image        = "2"
+    type         = "LINUX_CONTAINER"
+
+		environment_variable = {
+			"name"  = "SOME_KEY"
+			"value" = "SOME_VALUE"
+		}
+  }
+
+  source {
+    type     = "GITHUB"
+    location = "https://github.com/mitchellh/packer.git"
+  }
+
+  tags {
+    "Environment" = "Test"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
```

Comment:
Are all of these args optional except name? The schema suggests that's not the case